inspirehep/modules/workflows/models.py

```diff
@@ -74,3 +77,26 @@ class WorkflowsPendingRecord(db.Model):
         nullable=False,
     )
     record_id = db.Column(db.Integer, nullable=False)
+
+
+class WorkflowsRecordSources(db.Model):
+
+    __tablename__ = "workflows_record_sources"
+    __table_args__ = (
+        db.PrimaryKeyConstraint('record_id', 'source'),
+    )
+    source = db.Column(db.Text, default="", nullable=False)
+    record_id = db.Column(
+        UUIDType,
+        db.ForeignKey("records_metadata.id", ondelete='CASCADE'),
```

Comment:
This only allows for a 1-N relationship between records and workflow records. This is not the right relationship, because we might ingest things that will not become records, but are workflow records (we ingested them, but we also rejected them). This needs instead to be a M-N relationship with a connecting table.

See: https://en.wikipedia.org/wiki/One-to-many_(data_model) vs. https://en.wikipedia.org/wiki/Many-to-many_(data_model).